components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/js/JsAuthenticatedUser.java

```diff
@@ -18,175 +18,11 @@
-                return getLocalRoles();
-            case FrameworkConstants.JSAttributes.JS_CLAIMS:
-                if (StringUtils.isNotBlank(idp)) {
-                    return new JsRuntimeClaims(getContext(), step, idp);
-                } else {
-                    // Represent step independent user
-                    return new JsRuntimeClaims(getContext(), getWrapped());
-                }
-            default:
-                return super.getMember(name);
-        }
-    }
-
-    @Override
-    public void setMember(String name, Object value) {
-
-        switch (name) {
-            case FrameworkConstants.JSAttributes.JS_USERNAME:
-                getWrapped().setUserName((String) value);
-                break;
-            case FrameworkConstants.JSAttributes.JS_USER_STORE_DOMAIN:
-                getWrapped().setUserStoreDomain((String) value);
-                break;
-            default:
-                super.setMember(name, value);
-        }
-    }
-
-    @Override
-    public boolean hasMember(String name) {
-
-        switch (name) {
-            case FrameworkConstants.JSAttributes.JS_AUTHENTICATED_SUBJECT_IDENTIFIER:
-                return getWrapped().getAuthenticatedSubjectIdentifier() != null;
-            case FrameworkConstants.JSAttributes.JS_USERNAME:
-                return getWrapped().getUserName() != null;
-            case FrameworkConstants.JSAttributes.JS_USER_STORE_DOMAIN:
-                return getWrapped().getUserStoreDomain() != null;
-            case FrameworkConstants.JSAttributes.JS_TENANT_DOMAIN:
-                return getWrapped().getTenantDomain() != null;
-            case FrameworkConstants.JSAttributes.JS_LOCAL_CLAIMS:
-                return idp != null;
-            case FrameworkConstants.JSAttributes.JS_REMOTE_CLAIMS:
-                return idp != null && !FrameworkConstants.LOCAL.equals(idp);
-            default:
-                return super.hasMember(name);
-        }
-    }
-
-    private String[] getLocalRoles() {
-
-        if (idp == null || FrameworkConstants.LOCAL.equals(idp)) {
-            RealmService realmService = FrameworkServiceDataHolder.getInstance().getRealmService();
-            int usersTenantId = IdentityTenantUtil.getTenantId(getWrapped().getTenantDomain());
-
-            try {
-                String usernameWithDomain = UserCoreUtil.addDomainToName(getWrapped().getUserName(), getWrapped()
-                    .getUserStoreDomain());
-                UserRealm userRealm = realmService.getTenantUserRealm(usersTenantId);
-                return userRealm.getUserStoreManager().getRoleListOfUser(usernameWithDomain);
-            } catch (UserStoreException e) {
-                LOG.error("Error when getting role list of user: " + getWrapped(), e);
-            }
-        }
-        return ArrayUtils.EMPTY_STRING_ARRAY;
-    }
+public interface JsAuthenticatedUser  {
+    AuthenticatedUser getWrapped();
```

Comment:
new line before  method and method doc

Reply:
Changed and added method docs to similar interface methods